### DISCUSSION TITLE
Does regex search support multi-line?

### DISCUSSION THREAD

**Discussion by Ldoppea** [community/community#9278] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi,

I'm trying to search a code pattern that may be multi-line. I want to search some methods with specific parameter combination, but those parameters can be on one single line or on multiple lines. Having `\n` on my regex doesn't seems to work (or maybe my regex is wrong).

Is that scenario possible with the new search engine?

**Comment by gorzell** [comment, marked as answer]
Multi-line regex is supported.  For example if you have a query like [`/import os\n.*OptionParser/`](https://cs.github.com/?q=%2Fimport%20os\n.*OptionParser%2F) you get the expected results.  Although at the moment it looks like the highlighting might be off.

**Reply by Ldoppea** [reply]
Thank you for the answer. So it seems like my regex was wrong, or maybe I did not notice that the highlighting was off. I'll try again.

Thanks again 👍
